e2e-tests.sh

```diff
@@ -178,15 +177,15 @@ function initialize() {
   done
 
   if [[ "${CLOUD_PROVIDER}" == "gke" ]]; then
-      custom_flags+=("--addons=NodeLocalDNS")
+      extra_gcloud_flags+=("--addons=NodeLocalDNS")
   fi
```

Comment:
Maybe move this to `create_gke_test_cluster()` in `infra-library.sh` and drop `extra_gcloud_flags` env var here